buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -88,7 +89,18 @@ enum class ModuleType(val compatibility: JavaVersion) {
 }
 
 
+/**
+ * By default, we run an extra build step ("GRADLE_RERUNNER") which runs all test classes failed in the previous build step ("GRADLE_RUNNER").
+ * However, if previous test failures are too many (>10), this is probably not caused by flainess.
+ * In this case, we simply skip the GRADLE_RERUNNER step
+ */
+const val tooManyTestFailuresThreshold = 10
+
+
 class UnitTestAndCompilePlugin : Plugin<Project> {
+    private
+    val allTestFailuresCount = AtomicInteger(0)
```

Comment:
This is per project, isn't it? I think that is fine, though.

Reply:
Yes. Anyway we're running tests per project on CI, so that's fine.